core/src/OpenMP/Kokkos_OpenMP_Exec.hpp

```diff
@@ -213,7 +213,7 @@ void OpenMP::partition_master(F const& f, int num_partitions,
 namespace Experimental {
 
 template <>
-class MasterLock<OpenMP> {
+class KOKKOS_DEPRECATED MasterLock<OpenMP> {
```

Comment:
So GCC seems to be the only one complaining here.  No issue with Clang, Intel, nor MSVC.